pkg/deployment/deployment_item.go

```diff
@@ -85,11 +137,38 @@ func NewDeploymentItem(ctx SharedContext, project *DeploymentProject, collection
 		di.renderedYamlPath = filepath.Join(di.RenderedDir, ".rendered.yml")
 	}
 
-	err = di.Project.loadVarsList(di.VarsCtx, di.Config.Vars)
+	// update context vars with project/parrent path,references and tags
+	div := NewDeploymentItemVars(di.Project, di.Config, di.dir)
```

Comment:
Can you put building of the project/item vars into a helper function of `DeploymentItem`?.